### PR TITLE
Support for ToF Vl53l1x

### PR DESCRIPTION
**Pull request by Jason2866**
Feature Request #9152
Since i dont have the sensor i could not check with a ESP32

## Checklist:
  - [x] The pull request is done against the latest dev branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR.
  - [x] The code change is tested and works on Tasmota core ESP8266 V.2.7.4.1
  - [ ] The code change is tested and works on core ESP32 V.1.12.2
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

File: tasmota/xsns_77_vl53l1x.ino [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+*/
+
+#ifdef USE_I2C
+#ifdef USE_VL53L1X
+/*********************************************************************************************\
+ * VL53L1X
+ *
+ * Source:
+ *
+ * I2C Address: 0x52
```
Comment (device111) [inline tasmota/xsns_77_vl53l1x.ino:27]: should 0x29?

File: tasmota/xsns_77_vl53l1x.ino [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+\*********************************************************************************************/
+
+#define XSNS_77     77
+#define XI2C_53     53  // See I2CDEVICES.md
+
+#include <Wire.h>
+#include "VL53L1X.h"
+VL53L1X sensor;
```
Comment (device111) [inline tasmota/xsns_77_vl53l1x.ino:35]: object copy "sensor" should coverd with sensor Name "vl53l1x".

File: tasmota/xsns_77_vl53l1x.ino [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+uint8_t vl53l1x_ready = 0;
+uint16_t vl53l1x_distance = 0;
+/********************************************************************************************/
+
+void Vl53l1Detect(void) {
+  if (!I2cSetDevice(0x29)) { return; }
+
```
Comment (device111) [inline tasmota/xsns_77_vl53l1x.ino:43]: for better reading, define a var for the absolut I2C address.

File: tasmota/xsns_77_vl53l1x.ino [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+#define XSNS_77     77
+#define XI2C_53     53  // See I2CDEVICES.md
+
+#include <Wire.h>
+#include "VL53L1X.h"
+VL53L1X sensor;
+
+uint8_t vl53l1x_ready = 0;
+uint16_t vl53l1x_distance = 0;
+/********************************************************************************************/
```
Comment (device111) [inline tasmota/xsns_77_vl53l1x.ino:39]: you should  use structs for the global Vars.

**Comment by Jason2866** [comment]
Thx for your tips and comments. Since i am bloody noob in programming.
I would highly appreciate to do the changes you suggested.
Please do a PR on this forked branch. I will merge.
Thx a lot!

**Comment by device111** [comment]
I do directly a Pull Request here. it's easier...
@arendst  
Theo must have a look on it...

**Comment by Jason2866** [comment]
@device111  thx for optimizing. Closing as there is #9177 now

**Pull request closed without merging**